scripts/bc-wp-init.sh

```diff
@@ -41,19 +41,32 @@ nextip(){
     echo "$NEXT_IP"
 }
 
-# 
-if [[ -n `cat /etc/hosts | grep $domain` ]]
+# WIndows and Linux have diff paths to hosts file
+case "$OSTYPE" in 
+    linux-gnu*)  hosts_file="/etc/hosts" ;;
+    darwin*)  hosts_file="/etc/hosts" ;;
+    msys)  
+        # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+        hosts_file="C:\Windows\System32\drivers\etc\hosts"
+        ;;
+    *)
+        echo "OS could not be determined, please stall the installation of local development environment and contact DevOps"
+        exit 1
+        ;;
+esac
+
+if [[ -n `cat $hosts_file | grep $domain` ]]
 then
-    ip=`cat /etc/hosts | grep $domain | awk ' { print $1} ' | uniq`     
+    ip=`cat $hosts_file | grep $domain | awk ' { print $1} ' | uniq`
 else
-    max_used_ip=`cat /etc/hosts | grep -E '^192.168.33' | awk ' { print $1} ' | sort | uniq | tail -n 1`
+    max_used_ip=`cat $hosts_file | grep -E '^192.168.33' | awk ' { print $1} ' | sort | uniq | tail -n 1`
     test -n $max_used_ip && ip=$(nextip $max_used_ip) || ip="192.168.33.10"
 fi
 
 echo "Address $ip will be used for new vagrant box"
 sed -i -e "/^ip:/s/.*/ip: $ip/" site.yml
 
-forwarded_port=`python -c "import random; print random.randint(8000,8999)"`
+forwarded_port=`shuf -i 8000-8999 -n 1`
```

Comment:
Are `shuf` supported in all operating systems?

Reply:
Also, there is a chance that the port, when randomly chosen, will be the same as an existing port. Right?

Reply:
> Are `shuf` supported in all operating systems? I reckon it only works for Linux

shuf works on systems which I had a possibility to test - Ubuntu and Windows. shuf is part of coreutils, odds to have coreutils on MacOs very high. So I'm sure we need to use this.
 

> Also, there is a chance that the port, when randomly chosen, will be the same as an existing port. Right?

It can be, but firstly the probability is low, then checking the fact that another virtualbox instance took the same value for forwarded_port will complicate script (and even won't solve this problem completely). So I don't have another solution here.

Reply:
Firstly, `shuf` is not supported in macos, and Mac users, which is the main part of our developers, have to install `brew` and then `coreutils` through `brew` to use `shuf`. We can not ask developers cross-office to this, as it will complicate the process and we want the opposite. Please check this before deciding on a util. Secondly, why will it not solve the issue to check if the port is in use already? Use `netstat`, `ss` or even `vagrant port [id]` for this - I can reckon it can be resolved in a simple way